### PR TITLE
[minor] Add Workspace config generator

### PR DESCRIPTION
**Pull request by shajeena**
Implements a new role (`gencfg_workspace`) that is able to generate a Workspace custom resource and save the file to local disk.

Supports standard set of variables to configure the workspace:

- `mas_instance_id`
- `mas_workspace_id`
- `mas_workspace_name`

This allows us to remove the need for a user to pass in a pre-created workspace CR into the `mas_config_dir`.  Existing playbooks will be updated to take advantage of this capability, but a user does not need to use this, they can still pass a pre-created workspace custom resource directly to `suite_config`.  This just provides an alternative.

### CODE REVIEW

**Comment by shajeena** [comment]
@durera - Can you please review the changes

File: ibm/mas_devops/roles/suite_install/tasks/airgap-catalogs.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
```
Comment (durera) [inline ibm/mas_devops/roles/suite_install/tasks/airgap-catalogs.yml:1]: - Anything airgap related should go into the mas_airgap extension that is being worked on.
- Catalog installation is performed in ocp_setup_mas_deps role already

File: ibm/mas_devops/roles/suite_install/templates/airgap-catalogs.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (durera) [inline ibm/mas_devops/roles/suite_install/templates/airgap-catalogs.yml.j2:1]: - Anything airgap related should go into the mas_airgap extension that is being worked on.
- Catalog installation is performed in ocp_setup_mas_deps role already

File: ibm/mas_devops/roles/suite_config/tasks/main.yml [hunk 3]
```diff
@@ -48,3 +73,13 @@
     state: present
     definition: "{{ lookup('template', item.path) }}"
   with_items: "{{ mas_config_files.files }}"
+
```
Comment (durera) [inline ibm/mas_devops/roles/suite_config/tasks/main.yml:76]: If want to have a role create a workspace let's put it in a seperate new role please, e.g. `mas_workspace_create` ... this role's job is to apply configurations from a directory.  If one wants to use this to create a workspace they create the yaml for the workspace and pass the directory containing that file into the role (as we do today in daily FVT).

File: .idea/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (durera) [inline .idea/.gitignore:1]: These .idea files should be added to .gitignore and removed from the repo :+1:

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/suite_install/tasks/airgap-catalogs.yml (hunk 1 above)
Comment (shajeena) [reply ibm/mas_devops/roles/suite_install/tasks/airgap-catalogs.yml]: @durera - Thanks for the review 
Have removed the airgap extension from suite_install

File: ibm/mas_devops/roles/suite_install/templates/airgap-catalogs.yml.j2 (hunk 2 above)
Comment (shajeena) [reply ibm/mas_devops/roles/suite_install/templates/airgap-catalogs.yml.j2]: Have removed the airgap extension from suite_install

File: ibm/mas_devops/roles/suite_config/tasks/main.yml (hunk 3 above)
Comment (shajeena) [reply ibm/mas_devops/roles/suite_config/tasks/main.yml]: Created a new role suite_config_workspace and added the role in install_suite.yml . Please let me know if this is correct .
@santoshjpawar - FYI

File: .idea/.gitignore (hunk 4 above)
Comment (shajeena) [reply .idea/.gitignore]: Wrong checkin :) , thanks removed it

**Comment by shajeena** [comment]
@durera - can you please review comments addressed and approve the PR.Thanks

File: ibm/mas_devops/roles/suite_config/tasks/main.yml [hunk 5]
```diff
@@ -1,17 +1,41 @@
...
+  when: mas_workspace_id is not defined or mas_workspace_id == ""
+  fail:
+    msg: "mas_workspace_id property is required"
+
 - name: "Fail if mas_config_dir is not provided"
   when: mas_config_dir is not defined or mas_config_dir == ""
   fail:
     msg: "mas_config_dir property is required"
 
+# Wait for Mongocfg CRD
```
Comment (durera) [inline ibm/mas_devops/roles/suite_config/tasks/main.yml:20]: We can't assume the mongocfg is being applied here.  We do need something like this, but it needs to be more generic and able to handle the different xCfg resources that might (or might not) be passed into this role

**Review by durera — CHANGES_REQUESTED** [review]
See review comments.  I'll need to make some adjustments to the larger pipelines to support the changes in this PR before it's merged.  Will look to do that tomorrow.

File: ibm/mas_devops/roles/suite_config/tasks/main.yml (hunk 5 above)
Comment (shajeena) [reply ibm/mas_devops/roles/suite_config/tasks/main.yml:20]: @santoshjpawar - Can you please on this comment.

**Review by durera — APPROVED** [review]

**Pull request merged**